datafusion/common/src/scalar.rs

```diff
@@ -1716,7 +1716,11 @@ impl ScalarValue {
         } else {
             Self::iter_to_array(values.iter().cloned()).unwrap()
         };
-        Arc::new(array_into_list_array(values))
+        if values.len() <= i32::MAX as usize {
```

Comment:
I think the ScalarValue type needs to match the underyling array type -- so with this change, a `ScalarValue::List` might return `LargeListArray` or `ListArray`. This mismatch will likely cause issues downstream

I think the standard pattern would be to add a new function `ScalarValue::new_large_list` and `ScalarValue::LargeList` if they don't already exist